spec/lib/tasks/export/contracts_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'rails_helper'
+
+RSpec.describe 'rake export:contracts', type: :task do
+  let!(:complete_submission) do
+    create :submission,
+           aasm_state: 'completed',
+           framework: create(:framework, short_name: 'RM3786')
+  end
+
+  let!(:contract) do
+    # Explicit times are necessary because Export::Contracts::Extract.all_relevant
+    # relies on order(:created_date)
+    create :contract_entry, :legal_framework_contract_data, submission: complete_submission,
+                                                            created_at: Time.zone.local(2018, 12, 25, 13, 55, 59)
+  end
+
+  let!(:contract2) do
+    create :contract_entry, submission: complete_submission,
+                            created_at: Time.zone.local(2018, 12, 25, 14, 55, 59)
+  end
+
+  let(:extracted_contracts) { Export::Contracts::Extract.all_relevant }
+
+  context 'no args are given' do
+    let(:output_filename) { '/tmp/contracts_2018-12-25.csv' }
+    let(:args)            { {} }
+    let(:output_lines)    { File.read(output_filename).split("\n") }
+
+    around(:example) do |example|
+      travel_to(Date.new(2018, 12, 25)) { example.run }
+    end
+
+    before do
+      expect(contract.created_at).to be < contract2.created_at
```

Comment:
feels a bit weird having an expectation in the `before` block. is this also redundant because of the other expectations that check `output_lines[]`?

Reply:
It's a precondition that's saying "this test is order-dependent and if those `let` blocks above aren't respecting that I'm going to tell you".

Given that I'll stick a custom failure message on the end to make that (executably) explicit.